src/bindings/node/node.cc

```diff
@@ -213,6 +221,158 @@ class Actor {
     return outStr;
   }
 
+  static napi_value Locate(napi_env env, napi_callback_info info) {
```

Comment:
There's quite a bit of repetition in each of these wrapper functions. Did you look into whether we could take advantage of [macros](http://www.cplusplus.com/doc/tutorial/preprocessor/) or some other construct to reduce repeated code?

Reply:
i think you can do this pretty easily with lambdas and `std::function` (which is in `#include <functional>`):

```
  static napi_value Locate(napi_env env, napi_callback_info info) {
    return generic_action(env, info, [](actor_t* actor, const std::string& request) -> std::string {
        return actor->locate(request);
      }); // <---- this feels pretty javascripty amiright!?
  }

  //TODO: all the other functions should look basically just like this ^^

  static napi_value generic_action(napi_env env, napi_callback_info info, 
      const std::function<std::string(actor_t*, const std::string&)>& func) {
    napi_value jsthis;
    napi_status status;

    std::string reqString = ParseRequest(env, info, &jsthis);

    Actor* obj;
    status = napi_unwrap(env, jsthis, reinterpret_cast<void**>(&obj));
    checkNapiStatus(status, env, "Failed to unwrap js object");

    std::string locate_json;
    try {
      locate_json = func(actor, reqString);
    } catch (const std::exception& e) {napi_throw_error(env, NULL, e.what()); }

    auto outStr = WrapString(env, locate_json);
    return outStr;
  }
```

Reply:
gave it a shot with a few edits to satisfy the compiler. hopefully it works!